### PR TITLE
Move tablet management code from StorageEngine to TabletManager

### PR DESCRIPTION
**Pull request by yiguolei**
Move tablet mangement code from StorageEngine to TabletManager. 
The upper services such as clone, schema change are using StorageEngine to add tablet, delete tabet and StorageEngine uses tablet manager to finish the task currenlty. 
The upper services will use tablet manager directly if they only refers to the table related task in the future.

### CODE REVIEW

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset_meta_manager.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
[the remark this answers is not in the document]
Comment (yiguolei) [reply be/src/olap/rowset_meta_manager.h]: it is rowset meta manager, but I have not used it yet. I will remove it.

File: be/src/olap/store.cpp [hunk 2]
```diff
@@ -437,18 +437,19 @@ OLAPStatus OlapStore::deregister_tablet(Tablet* tablet) {
...
 }
 
-std::string OlapStore::get_shard_path_from_header(const std::string& shard_string) {
+std::string OlapStore::get_absolute_shard_path(const std::string& shard_string) {
     return _path + DATA_PREFIX + "/" + shard_string;
 }
 
-std::string OlapStore::get_tablet_schema_hash_path_from_header(TabletMeta* header) {
-    return _path + DATA_PREFIX + "/" + std::to_string(header->shard())
+std::string OlapStore::get_tablet_path_from_header(TabletMeta* header, bool with_schema_hash) {
```
[the remark this answers is not in the document]
Comment (yiguolei) [reply be/src/olap/store.cpp]: I rename it to get_absolute_tablet_path

File: be/src/olap/tablet_manager.h [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            TTabletId tablet_id, SchemaHash schema_hash,
+            Version version, VersionHash version_hash);
+
+    OLAPStatus create_tablet(const TCreateTabletReq& request, 
+                             std::vector<OlapStore*> stores);
+
+    // Create new tablet for StorageEngine
+    //
+    // Return Tablet *  succeeded; Otherwise, return NULL if failed
+    TabletSharedPtr create_tablet(const TCreateTabletReq& request,
```
Comment (chaoyli) [inline be/src/olap/tablet_manager.h]: OLAPStatus is used to indicate status like disk error, and so on. So return a TabletSharedPtr may can not include this error.

Comment (yiguolei) [reply be/src/olap/tablet_manager.h]: StorageEngine has two create method， TabletManager has two related method. And I will remove the duplcate method when refactoring upper services.

**Pull request merged**